src/rust/engine/src/intrinsics.rs

```diff
@@ -292,19 +303,23 @@ fn remove_prefix_request_to_digest(
 ) -> BoxFuture<'static, NodeResult<Value>> {
   let core = context.core;
   let store = core.store();
-
   async move {
-    let input_digest =
-      lift_directory_digest(&externs::getattr(&args[0], "digest").unwrap()).map_err(throw)?;
-    let prefix: String = externs::getattr(&args[0], "prefix").unwrap();
-    let prefix = RelativePath::new(PathBuf::from(prefix))
+    let (input_digest, raw_prefix) = Python::with_gil(|py| {
+      let py_remove_prefix = args[0].clone_ref(py).into_ref(py);
+      let py_digest = externs::getattr(py_remove_prefix, "digest").unwrap();
+      let input_digest = lift_directory_digest(py_digest).map_err(throw)?;
+      let prefix: String = externs::getattr(py_remove_prefix, "prefix").unwrap();
+      let res: NodeResult<(Digest, String)> = Ok((input_digest, prefix));
+      res
+    })?;
```

Comment:
This pattern of returning a tuple is so that we can consolidate all GIL access into one closure. I use it throughout most this file.